### PR TITLE
update to angular2 release version

### PR DESCRIPTION
**Pull request by Vrolijkx**
### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

This is a big one with all the changes needed to get up to date with the released angular2. It's build on the work of @sebastianhaas to get the modules/imports working correctly. I tweaked the code so it's easy consumable in any client project. The next changes happened in this pull request
- replace ```* as models``` import with multiple imports.
- Added a ngModule template with a forConfiguration creation method.
- Added basePath to the configuration object
- Added an easy constructor to the configuration object
- Updated versions in ```package.json```
- Some additional tweeks

### CODE REVIEW

File: modules/swagger-codegen/src/test/java/io/swagger/codegen/AbstractIntegrationTest.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
     protected abstract Map<String, String> configProperties();
 
     // @wing328: ignore for the time being until we fix the error with the integration test
-    @Test(enabled = false)
+    @Test(enabled = true)
```
Comment (Vrolijkx) [inline modules/swagger-codegen/src/test/java/io/swagger/codegen/AbstractIntegrationTest.java]: @wing328 can you check again if these tests are still a problem? They run fine on my machine an i think they have a nice added value.

**Comment by sebastianhaas** [comment]
Many thanks for polishing this up for a PR, I've been quite busy lately. Looks good to me.

**Comment by wing328** [comment]
@Vrolijkx thanks for the PR. Please use 4-space instead of tab for code indentation in Java files.

Please also rebase on the latest mas

**Comment by wing328** [comment]
The CI (Travis) failed due to Android (volley) Petstore API client: https://api.travis-ci.org/jobs/177505060/log.txt?deansi=true

I would recommend you to rebase on the latest master to resolve it.

File: modules/swagger-codegen/src/test/java/io/swagger/codegen/AbstractIntegrationTest.java (hunk 1 above)
Comment (wing328) [inline modules/swagger-codegen/src/test/java/io/swagger/codegen/AbstractIntegrationTest.java]: If my memory serves me well, the test failed to run on Windows and that's why I commented it out for the time being until some Windows developers have a chance to fix it.

**Comment by Vrolijkx** [comment]
Ok, but I started from 2.3.0 branch because it's a breaking change. So don't I need to rebase on the latest 2.3.0 branch?

**Comment by wing328** [comment]
Oh yes, please rebase on the latest 2.3.0 branch instead.

**Comment by Vrolijkx** [comment]
@wing328 I get a lot of errors while trying to rebase this fix. Probably because I already merged 2.3.0 and master in this branch. Can you maybe take a look?

**Comment by wing328** [comment]
@Vrolijkx sure I'll take a look. I may create another PR by cherry picking the commits you submitted in this PR.

**Comment by Vrolijkx** [comment]
@wing328 No problem. Thanks in advance.

**Comment by wing328** [comment]
Closed via #4272

**Pull request closed without merging**